### ISSUE TITLE
Discussion of project's future, forks, etc.

### ISSUE THREAD

**Issue by alphapapa** [alphapapa/Particle#1]
Regarding the discussion at https://github.com/ParticleCore/Particle/issues/528

Given that the original author has sold out his users to a hostile party with malicious intent, he and his work can no longer be trusted in any form.  The original repo must be blacklisted and the author's work must henceforth be rejected.  This includes his userscripts and the Iridium browser extension, which he purports to continue to control.

I am not expert enough in JavaScript, browser internals, etc. to maintain this project going forward, but I offer this repo and issue as a temporary home for interested users and developers to discuss the project's future, sans "ParticleCore" (whoever that actually is).

Issues going forward:

1.  The license is unclear.  It would be good to dig into the repo history and see if he has copied any code from other projects which might carry their own licenses (which, if so, he omitted).  Such parts could be used without regard for his own license (or lack thereof).
2.  If we can fork the project, the name should be changed.
3.  Regardless of those items, this issue should be publicized to protect existing users of the extension and anyone who might use this author's work in the future.  He can no longer be trusted, and no one should install or use his software.
4.  If we cannot fork the project for license or other reasons, we should try to identify the best alternative projects and recommend them to users.

Archived copies of the issues on the original repo as of now:

http://archive.is/VxYeE
http://archive.is/xZXQG

Updated:

http://archive.is/FRATi

**Comment by ColonelGerdauf** [comment]
While I appreciate what you are trying to do, I personally refuse to support this fork due to (asides from the obvious EOL status) your needlessly hostile and judgemental attitude towards the original repository owner. It is quite frankly childish. 

Did ParticleCore make big mistakes? Yes; in fact this was not his first. 
Are they as 'unforgivable' as you make it sound? Not really.
Do you have a right to his identity? Absolutely not!

In your conversations with him, you have conveniently ignored details of what he has typed out in plain English, and used it to spew conjectures and flat-out inaccuracies about what has been happening.  He did not "sell out" anyone to anyone. Let me ask you; did you know about the wrongdoings of the buyer before the incident? Have you even heard of the buyer? No, you did not. Could you do research on the buyer assuming you knew the name? Probably, but prior to the incident would have netted very little; that is how these malicious parties work. You then have the audacity to call out on his use of a nickname (which is a pointless argument on its own), when you are doing the same yourself.

On top of that, let me repeat that this is an EOL product. Silent transfers of ownership of brands/products near or at EOL is commonplace in the tech world. NDAs are very tricky and complex monsters; do not take that so idealistically. This is something that is already deemed 'deprecated'. He has nothing going forward with this project. If was to sell out, he would have already transferred ownership of the other extensions/addons, but he did not. In the case of Chrome, ParticleCore got duped; he is only human at the end of the day.

To answer your issues (I have checked the code myself):

1) There is no borrowed code or external API anywhere. He may have used a few abstract ideas for inspiration, and he has accepted pull requests from other forks, but those do not warrant a license. 
2) And this achieves what, exactly? The product is EOL; this will only generate needless confusion as the main repository contains no malicious code to begin with.
3) As I had explained earlier, you are stretching the truth with that one; so much so that I am now very suspicious of your political motives.
4) Sure, you can do that. You have the power to provide a recommendation as you see fit, but do not pester others that have made up their minds for better or for worse.

And before you say it, I have nothing to do with ParticleCore myself. To me, he is a stranger as are you. I am not being "paid" to say this, as I am speaking my mind on this matter that you have blown out of proportion and taken out of context. I understand that trust is fickle, but that does not ignore the fact that you need to look at the bigger picture.

**Comment by alphapapa** [comment]
@ColonelGerdauf Your comment is mostly just mildly amusing.  You tear down some strawmen, make some poor judgments in the same way that the author did, and then accuse me of "political motives", which is laughable since, if either of us had any "political motives," your emotionally loaded comment would show that it's you.

> your needlessly hostile and judgemental attitude towards the original repository owner. It is quite frankly childish.

No, it's not.  Read all of the comments I wrote.  Considering it one's duty to take users' safety and security seriously is not childish, it's mature.  If you think it is, you're calling good "evil" and evil "good."

> Are they as 'unforgivable' as you make it sound? Not really.

Note carefully that he has not repented; in fact, he boldly proclaimed that *he will do it again if someone offers him a similar deal.*  Forgive him if you want, that's your personal decision.  No one else is under any obligation to naively offer their computer systems to him as fodder for his wallet, building up the "Downloads" number so he can sell out again in the future.  Again, he explicitly said that he *will* do that.  Are you nuts?

> In your conversations with him, you have conveniently ignored details of what he has typed out in plain English, and used it to spew conjectures and flat-out inaccuracies about what has been happening.

That's a lie.  Everything I said turned out to be 100% true; ParticleCore confirmed it.

> He did not "sell out" anyone to anyone.

That's a lie.  He admitted to doing it.

> Let me ask you; did you know about the wrongdoings of the buyer before the incident? Have you even heard of the buyer? No, you did not. Could you do research on the buyer assuming you knew the name? Probably, but prior to the incident would have netted very little; that is how these malicious parties work. 

I don't know who the buyer is now.  No one does except ParticleCore, if even he does.  That's one of the points.

Another is that he's not an Internet noob.  He's been making browser extensions for years.  He knows that they get hijacked and have malware and adware installed.  My TabTint browser extension is a fork of a Firefox extension that the same thing happened to *several years ago*: the author sold out, the new owner installed malware that *injected JavaScript from superfish.com into pages that users visited*, and Mozilla did nothing about it.  When I discovered that, I ripped out the malware and forked it into a new extension.

There's no excuse for him not realizing what was going to happen.  He chose money for himself over the safety and good of other people.

> You then have the audacity to call out on his use of a nickname (which is a pointless argument on its own), when you are doing the same yourself.

That's another lie.  You can find my name and email address all over my repos.  You haven't done your homework.  

> On top of that, let me repeat that this is an EOL product. Silent transfers of ownership of brands/products near or at EOL is commonplace in the tech world.

Blah blah blah, meaningless "tech world" gibberish and hand-waving.

> NDAs are very tricky and complex monsters; do not take that so idealistically.

No one signs an NDA at gunpoint.  He willingly submitted himself to a contract that prohibits him from exposing the misdeeds of the guy who paid him a bucket of money to install malware on unsuspecting, trusting people's computers.  That's making a deal with the devil.  No excuses.

> If was to sell out, he would have already transferred ownership of the other extensions/addons, but he did not.

Another lie from you.  He hasn't sold them out because he hasn't received any offers.  He *explicitly said that he will do it again when he gets an offer.*

>  In the case of Chrome, ParticleCore got duped; he is only human at the end of the day.

"Gee, honest officer, I didn't know the goods were stolen.  He seemed like such a nice guy!  Yeah, I mean, he told me not to tell anyone where I got them from, but I didn't think anything of it."  Please.

> There is no borrowed code or external API anywhere. He may have used a few abstract ideas for inspiration, and he has accepted pull requests from other forks, but those do not warrant a license.

Oh, well, since you "checked the code" yourself, you must be correct.  Come on, how would you know?  Did you google for every line of code?  I don't know why you're even bothering to say this.

> And this achieves what, exactly? The product is EOL; this will only generate needless confusion as the main repository contains no malicious code to begin with.

Finally you reveal your own knee-jerkiness.  If you had actually read the comment where I linked to this repo, you'd know why I posted this, and you wouldn't have gone flying off the handle like this.

> As I had explained earlier, you are stretching the truth with that one; so much so that I am now very suspicious of your political motives.

Another lie.  I never claimed the original repo contained malicious code.  If I had, I'd be an idiot, since I forked this one from its head commit.

Is this your Internet Good Deed of the Day, swooping in on a weeks-old controversy on a page no one else has even seen before (no one has commented here in 20 days since I posted it) to defend a guy you know nothing about?  Poor (not financially poor now!) little browser extension author who got criticized for selling to a shadowy, secretive little business in who-knows-where so they could sneak malware into his users' browsers?  Gee, what a swell guy you are for coming here and standing up for him.

Please.  What are *your motives*?

> Sure, you can do that. You have the power to provide a recommendation as you see fit, but do not pester others that have made up their minds for better or for worse.

It's a free Internet.  I will say what I want, and you can do the same, as you are yourself.

> And before you say it, I have nothing to do with ParticleCore myself. To me, he is a stranger as are you. I am not being "paid" to say this, as I am speaking my mind on this matter that you have blown out of proportion and taken out of context. I understand that trust is fickle, but that does not ignore the fact that you need to look at the bigger picture.

I seriously doubt the veracity of your words.  Your emotional reaction suggests that you do indeed have something invested of yourself in this matter.  If you are now being honest, then you have wildly overreacted without even understanding what you are reacting to.  It is you who needs to look at the bigger picture, and it is you who needs to do his homework, read the entire threads on the ParticleCore repo, and see for yourself that he *will do this again when he gets an offer.*

**Comment by ColonelGerdauf** [comment]
For all of your verbal aggression and ego, your response only proves that you are not willing to listen to other people's commentary, continuing your slant on matters where they have already been put to rest, and attacking others for questioning it. Calling me a liar at every comment I made (of which you have conveniently quote-mined) does not prove anything; it only weakens your argument. 

> No, it's not. Read all of the comments I wrote. Considering it one's duty to take users' safety and security seriously is not childish, it's mature. If you think it is, you're calling good "evil" and evil "good."
I have already done so, otherwise I would not have bothered responding. That is a very immature straw-man case example you are pulling, which once again shows your refusal to listen. Yes, it is a very good thing to take users security seriously, that is not my issue; my issue is how you approach the matter. 

> Note carefully that he has not repented; in fact, he boldly proclaimed that he will do it again if someone offers him a similar deal. Forgive him if you want, that's your personal decision. No one else is under any obligation to naively offer their computer systems to him as fodder for his wallet, building up the "Downloads" number so he can sell out again in the future. Again, he explicitly said that he will do that. Are you nuts?
Are you? He has apologized in the best way he thought he could, when it was the fist time it had occurred. Don't be dense; that is not what he was saying. What he was saying is that he is not in avoidance of the possibility of another deal (and he meant 'deal' in a very broad sense, which flew over your head). He only sold a product that was already dead. That does not mean that he will do the same thing again. False association.

> Another is that he's not an Internet noob. He's been making browser extensions for years. He knows that they get hijacked and have malware and adware installed. 
So what if he is not an internet noob? That does not make him a deity by any stretch of the imagination. People can make some really stupid mistakes; that is humanity in a nutshell. Also, he may know all he wants, but when somebody makes a deal and they had been checked out as clean,

> You can find my name and email address all over my repos. You haven't done your homework.
This was not the point of your original argument, and this is only being evasive. Also, I have to do quite a bit of digging to find this information, so you have made a moot point here.

> Blah blah blah, meaningless "tech world" gibberish and hand-waving.
You can call it what you want. It is the cold reality of the world we live in, so it is odd that you pick on one person for doing something everybody else has been doing (in principle) for decades now.

> No one signs an NDA at gunpoint. He willingly submitted himself to a contract that prohibits him from exposing the misdeeds of the guy who paid him a bucket of money to install malware on unsuspecting, trusting people's computers. That's making a deal with the devil. No excuses.
He was given a promise from his own demands in turn; a promise that did not pan out as he hoped. He could not have known that it was malware; hindsight is only 20/20. The "no excuses" attitude can only get you so far in life.

> Oh, well, since you "checked the code" yourself, you must be correct. Come on, how would you know? Did you google for every line of code? I don't know why you're even bothering to say this.
Because I know Javascript. It is not too hard for me to tell what code is original or not. An API call can be found by the use of noted keywords. His code is mostly a collection of variables, identifiers, and basic String manipulation. Don't be a douche; there are better ways to counter my arguments.

> Is this your Internet Good Deed of the Day, swooping in on a weeks-old controversy on a page no one else has even seen before (no one has commented here in 20 days since I posted it) to defend a guy you know nothing about? Poor (not financially poor now!) little browser extension author who got criticized for selling to a shadowy, secretive little business in who-knows-where so they could sneak malware into his users' browsers? Gee, what a swell guy you are for coming here and standing up for him.
If I had even cared about good deed publicity, I would have posted it in the repo discussion. I decided to post it here, hoping to address the matter with you directly. Like I said, I do not really care much about ParticleCore; I responded because I do not like your attitude specifically. Your could have responded to me at least in a more well-mannered approach, but with the way you are speaking right now, you have given me the figurative victory on a silver platter when I never even asked for it.

> It's a free Internet. I will say what I want, and you can do the same, as you are yourself.
Any person with a single brain cell knows that there are practical limits to this 'freedom' that you cherish so much. Freedom, even as a base concept, is not absolute; you know better than this. People are free to make their own decisions, but not at the cost of the freedom of the other people. 

> -everything else-
You are jumping the gun on accusations here, especially considering that it was not a repeat occurrence. Saying that I have "overreacted" is akin to the pot calling the kettle black. Your attitude is the problem, and you cannot use "freedom" as an excuse. I had offered a sincere response with what I had read from you and the others. You could have simply clarified your position in a neutral tone (like you should have in the original discussion), and perhaps I would have been more forgiving myself. Unfortunately, you have decided to attack me without offering a lot of sound justification for your point of view. In other words, you have shot yourself in the foot, and then blame me for it.

**Comment by alphapapa** [comment]
> For all of your verbal aggression and ego,

Your hypocrisy is stunning.  Your very first message to me included:

> your needlessly hostile and judgemental attitude

> It is quite frankly childish.

> you have conveniently ignored details of what he has typed out in plain English, and used it to spew conjectures and flat-out inaccuracies

> You then have the audacity

> you are stretching the truth with that one; so much so that I am now very suspicious of your political motives.

> you have blown out of proportion

And now *you* have the audacity to accuse *me* of verbal aggression.

> your response only proves that you are not willing to listen to other people's commentary

Another lie.  I "listened" to every word you wrote, considered them carefully, and responded in detail.  You simply don't like that I disagree with you, and instead of accepting the disagreement amiably, you attack me and insult me.

> continuing your slant

My "slant" is my opinion, and you have one as well.  It is hypocritical of you to imply that it's wrong for me to have one.

> on matters where they have already been put to rest

A meaningless, subjective statement.  "You can't talk about that anymore, it's been settled."  Are you an aspiring politican?  Trying to shut up people who disagree with you.  Shameful.

> and attacking others for questioning it

More hypocrisy from you.  *You* came *here* attacking *me*.

> Calling me a liar at every comment I made

I did not call you a liar--I called out the lies you told, because they were factually incorrect.

> (of which you have conveniently quote-mined)

More hypocrisy, as you are "quote-mining" me right now.

Meanwhile, on the Internet, and especially in the FOSS world, this is known as "replying inline," and it is standard practice.  But, yet again, you imply that I'm doing something wrong, while doing it yourself!  (And in a moment, you will pull out the pot-calling-kettle-black analogy!)

> does not prove anything

I'm not here to prove anything, and neither are you.  This is all a meaningless argument over *opinions.*

> it only weakens your argument.

For whom are you performing?  This is not a debate contest.

> > No, it's not. Read all of the comments I wrote. Considering it one's duty to take users' safety and security seriously is not childish, it's mature. If you think it is, you're calling good "evil" and evil "good."
> 
> That is a very immature straw-man case example you are pulling,

What strawman?  This is what actually happened: he chose money over protecting users.  You keep calling me "immature" while I'm advocating protecting others over one's own financial interests--in other words, maturity.

> which once again shows your refusal to listen.

You keep making this claim that is facially false.  You are conflating disagreement with refusal to listen.  How arrogant of you to insist that I must agree with you or else I am not hearing you.  It implies that you are necessarily correct, and anyone who disagrees with you is necessarily wrong.  *That* is childish and immature.  Here, have a quote:

> The test of a first rate intelligence is the ability to hold two opposed ideas in the mind at the same time, and still retain the ability to function.

--F. Scott Fitzgerald

> Yes, it is a very good thing to take users security seriously, that is not my issue; my issue is how you approach the matter.

It's none of your concern how I approach the matter.  If you don't like it, you are free to ignore me--this is the Internet.

> Don't be dense; that is not what he was saying. What he was saying is that he is not in avoidance of the possibility of another deal (and he meant 'deal' in a very broad sense, which flew over your head). That does not mean that he will do the same thing again. He only sold a product that was already dead.  False association.

Another lie from you.  This is a direct quote: "Make no mistake, if I am offered another deal I will take it again."  See for yourself: https://archive.is/o/FRATi/https://github.com/ParticleCore/Particle/issues/528%23issuecomment-315840535

> So what if he is not an internet noob? That does not make him a deity by any stretch of the imagination. People can make some really stupid mistakes; that is humanity in a nutshell.

Amazing that you are going out of your way to downplay his actions as merely a mistake, while harshly attacking me for criticizing that mistake which put many people's safety at risk.

You obviously have something invested here, and you should disclose your affiliation with the author or project.

> Also, he may know all he wants, but when somebody makes a deal and they had been checked out as clean, what would you do, realistically?

What is this, "checked out as clean"?  You are speculating about what happened between him and the company, something of which none of us have any information, because he has declined to explain any details, because of having signed a contract.

Or do you actually know things that you are not telling?  I am beginning to wonder.

> > You can find my name and email address all over my repos. You haven't done your homework.
> 
> This was not the point of your original argument

It certainly is!  My identity is not a secret here.  But where can I look up the name and email address of "ParticleCore"?  If I have overlooked it, please let me know.

> and this is only being evasive

It is nothing of the kind.  It is the truth.

> Also, I have to do quite a bit of digging to find this information

Preposterous.  Go to any of my pinned repos, open the source file, see the name and email address.  If you think that is "quite a big of digging," it is not surprising that you attacked me in the first place, based on your uninformed reactions.

> so you have made a moot point here.

Assertions do not an argument make.

> > Blah blah blah, meaningless "tech world" gibberish and hand-waving.
> 
> You can call it what you want. It is the cold reality of the world we live in, so it is odd that you pick on one person for doing something everybody else has been doing (in principle) for decades now.

You try to defend your meaningless hand-waving with more meaningless handwaving?  Wow.  Who is this "everybody else" you're talking about?  Gee, I guess if everyone was robbing banks, it'd be okay to do that too.  Hey, here's a thought: everyone on the Internet has opinions you don't agree with and says things you don't like--so isn't it odd that you pick on one person for doing something everybody else has been doing (in principle) for decades now?  Does your hypocrisy know any bounds?

> > No one signs an NDA at gunpoint. He willingly submitted himself to a contract that prohibits him from exposing the misdeeds of the guy who paid him a bucket of money to install malware on unsuspecting, trusting people's computers. That's making a deal with the devil. No excuses.
> 
> He was given a promise from his own demands in turn; a promise that did not pan out as he hoped. He could not have known that it was malware; hindsight is only 20/20.

How do you know what promise he was given?  None of us have seen the contract he signed, or the emails or phone calls he exchanged.  My doubts about your motivation continue to grow--or you're just continuing to jump to greater and greater conclusions.

> The "no excuses" attitude can only get you so far in life.

What does this even mean?

> > Oh, well, since you "checked the code" yourself, you must be correct. Come on, how would you know? Did you google for every line of code? I don't know why you're even bothering to say this.
> 
> Because I know Javascript. It is not too hard for me to tell what code is original or not.

This is honestly laughable.  Any IP lawyer would have a field day with you.

>  An API call can be found by the use of noted keywords.

Have you even heard of Oracle v. Google?  SCO v. IBM?  You are so uninformed, it's amazing.

> His code is mostly a collection of variables, identifiers, and basic String manipulation.

Like every piece of software connected to the Internet.  Yet all of it's covered by copyright, except for that explicitly released into the public domain.  There is a reason that, e.g. Debian takes the DFSG so seriously, and why IP and corporate lawyers around the world make millions of dollars litigating such cases.

> Don't be a douche; there are better ways to counter my arguments.

Physician, heal thyself!

> If I had even cared about good deed publicity, I would have posted it in the repo discussion. I decided to post it here, hoping to address the matter with you directly. Like I said, I do not really care much about ParticleCore; I responded because I do not like your attitude specifically.

If this is really your motivation, you must be a busy, busy person, attacking everyone on the Internet whose attitude you do not like.

> You could have responded to me at least in a more well-mannered approach

If you could only hear yourself!

> but with the way you are speaking right now, you have given me the figurative victory on a silver platter when I never even asked for it.

What victory?  This is not a contest!  This is a meaningless argument on the Internet.  Are you also unfamiliar with https://xkcd.com/386/ ?

> > It's a free Internet. I will say what I want, and you can do the same, as you are yourself.
> 
> Any person with a single brain cell knows that there are practical limits to this 'freedom' that you cherish so much. Freedom, even as a base concept, is not absolute; you know better than this.

Aha, now this starts to get interesting.  What are you insinuating here?  That I am not allowed to have an opinion different from yours?  That I am not allowed to express it in a way that you do not personally approve of?

And it's amazing that you accused me of strawmanning, when you're running with full-blown argumentum ad absurdum, talking about practical limits to freedom, as if I had argued that I should be allowed to punch you in the face.  Really, amazing.

>  People are free to make their own decisions, but not at the cost of the freedom of the other people.

First of all, that's wrong, because people make decisions that restrict others' freedom all the time.  When I decide where to park my car, it denies someone else the freedom to park there.  When I make an appointment with a doctor, it denies someone else the freedom to use that time slot.  When I register a domain name, it denies everyone else the freedom to register that domain name.  When I buy all the donuts at the grocery store, it denies everyone else the freedom to buy those donuts.

So what you're effectively arguing is that *your* freedom should be preserved at the expense of *my* freedom.  You want to make decisions that restrict *my* freedom, but you don't want anyone to restrict *your* freedom.

There's a word for that: authoritarian.  It's impressive that you think yourself worthy to discharge the duties of deciding who should have what freedoms.

> You are jumping the gun on accusations here,

That's incorrect.  My accusation was borne out by his own words.

> especially considering that it was not a repeat occurrence.

Irrelevant.  Trust comes with time, yet it is destroyed in a moment.  And he even warned us that he will do it again!

> Saying that I have "overreacted" is akin to the pot calling the kettle black.

Well then, you're as black as I am.  Where does that leave us?

> Your attitude is the problem

How is my attitude any of your concern?

> and you cannot use "freedom" as an excuse.

Are you for real?  You're arguing that *I am not free to determine my own attitude*?

> I had offered a sincere response with what I had read from you and the others.

Oh, it was definitely sincere.

> You could have simply clarified your position in a neutral tone

You could have offered *your* position in a neutral tone, but you began with personal attacks.  You've now called me "hostile," "judgmental," "childish," "dense," and "a douche."  And you accuse *me* of "verbal aggression"!  Amazing!

> and perhaps I would have been more forgiving myself.

What?  Who are you to judge me guilty and offer your personal forgiveness for my "attitude"?  Are you serious?

> Unfortunately, you have decided to attack me

This is just bizarre.  *20 days* after I clicked the "fork" button and *no one even noticed or cared*, *you* show up out of the blue and verbally attack *me*.  Now you're turning this around on me, claiming that I'm attacking you.  Are you delusional?

> without offering a lot of sound justification for your point of view.

Justification is in the eye of the beholder, and clearly you have appointed yourself judge, jury, and executioner.

> In other words, you have shot yourself in the foot, and then blame me for it.

You burst in the door, guns blazing, trying to put on some kind of a show for an imagined audience, and then accuse me of dragging you in and attacking you.

Bizarre.
